modules/test/playwright/tests/frontend-data-set-views-web/fields.spec.ts

```diff
@@ -0,0 +1,154 @@
+			await fieldsPage.addParentField('title');
+			await fieldsPage.addParentField('creator');
+			await fieldsPage.addChildField('creator', 'name');
+			await fieldsPage.addChildField('creator', 'id');
+		});
+
+		await test.step('Save changes', async () => {
+			await fieldsPage.saveAddFieldsModal();
+		});
+
+		await test.step('Fields are present in table', async () => {
+			await expect(page.getByText('dateCreated').first()).toBeVisible();
+			await expect(page.getByText('title').first()).toBeVisible();
+			await expect(page.getByText('creator.*').first()).toBeVisible();
+			await expect(page.getByText('creator.id').first()).toBeVisible();
+			await expect(page.getByText('creator.name').first()).toBeVisible();
+		});
+	});
+
+	test('Add DataSet fragment', async ({applicationsMenuPage, page}) => {
+		await test.step('Go to Home', async () => {
+			await applicationsMenuPage.goToHome();
+		});
+
+		await test.step('Click on "Edit" button', async () => {
+			const editPageButton = await page.getByRole('link', {
+				name: 'Edit',
+			});
+			await editPageButton.click();
+		});
+
+		await test.step('Search for "Data Set" fragment', async () => {
+			const fragmentSearchInput = await page.getByLabel(
+				'Search Fragments and Widgets'
+			);
+			fragmentSearchInput.fill('Data Set');
+		});
+
+		await test.step('Drag "Data Set" fragment & Drop into the page editor w/ keyboard', async () => {
+			const source = await page.getByRole('menuitem', {
+				name: 'Data Set Add Data Set Mark Data Set as Favorite',
+			});
+			await source.focus();
+			await source.press('Enter');
+			await source.press('Enter');
+		});
+
+		await test.step('Select empty Data Set fragment', async () => {
+			await page
+				.getByText('Select a data set view. Beta')
+				.first()
+				.click();
+		});
+
+		await test.step('Open Data Set View Selector', async () => {
+			await page
+				.getByRole('button', {name: 'Select Data Set View'})
+				.click();
+			await page
+				.getByRole('menuitem', {name: 'Select Data Set View...'})
+				.click();
+		});
+
+		await test.step('Select Data Set View', async () => {
+			await expect(page.getByRole('dialog')).toBeVisible();
+			await expect(
+				page.getByRole('heading', {name: 'Select'})
+			).toBeVisible();
+			await page
+				.frameLocator('iframe[title="Select"]')
+				.locator('li')
+				.filter({hasText: 'Data Set View Sample'})
+				.first()
+				.click();
+			await page
+				.frameLocator('iframe[title="Select"]')
+				.getByRole('button', {name: 'Save'})
+				.click();
+		});
+
+		await test.step('Publish page with Data Set View', async () => {
+			await page.getByRole('button', {name: 'Publish'}).click();
+
+			await applicationsMenuPage.goToHome();
+
+			await expect(
+				page.locator('.data-set-wrapper').first()
+			).toBeVisible();
+		});
+
+		await test.step('Fields are present in fragment table heading', async () => {
+			await expect(
+				page.getByRole('button', {name: 'creator.*'})
```

Comment:
A potential improvement for next iterations is to model the notion of "A page with the DS fragment in it" / "A page with the FDS in it" as a dedicated POM. This way, all interactions and expectations we code over the FDS would be in a single place, where some useful wrappers (like `assertFieldIsPresent(fieldName)` in this particular example) would be used by tests conveniently

Again not for this PR but just a consideration

Reply:
I'd love this approach!